practice-app/tests/test_worldtime.py

```diff
@@ -0,0 +1,102 @@
+from app.views import User
+from app.worldtime import WorldTimeTable
+
+TEST_USER = "practiceapp"
+TEST_USERPASS = "practiceapp"
+
+def test_signup(client):
+    client.post('/signup', data=dict(
+        username=TEST_USER, password=TEST_USERPASS, password_confirm=TEST_USERPASS
+    ), follow_redirects=True)
+    user = User.query.filter_by(username=TEST_USER).first()
+    assert str(user.username) == TEST_USER
+
+
+def test_login(client):
+    client.post("/login", data=dict(
+        username=TEST_USER, password=TEST_USERPASS
+    ), follow_redirects=True)
+    response = client.get("/")
+    assert response.status_code == 200
+
+
+def test_invalid_login(client):
+    client.post("/login", data={"username": "not_logged_username", "password": "not_logged_username_password"})
+    response = client.get("/")
+    assert response.status_code == 302
+
+
+def test_worldtime_get(client):
+    client.post("/login", data=dict(
+        username=TEST_USER, password=TEST_USERPASS
+    ), follow_redirects=True)
```

Comment:
You don't need to login each time. Since it logins once at the start of tests, you can continue with the logged user.